### ISSUE TITLE
Instant crash issue

### ISSUE THREAD

**Issue by haashemi** [MikeModder/anpan#11]
Hi, Bot gets instant crash on any new message with added prefix.

Go version: 1.16.4
discordgo version: v0.23.2
anpan version: v0.0.0-20210425181706-21d818f4ab3f (it must be the latest one)

Code:

```
func main() {
	client, err := discordgo.New...

	handler := anpan.New([]string{","}, []string{"MyID"}, true, true, client.StateEnabled)
	// from /example
	handler.AddCommand("ping", "Check the bot's ping.", []string{"pong"}, false, false, discordgo.PermissionSendMessages, discordgo.PermissionSendMessages, anpan.CommandTypeEverywhere, pingCommand)

	client.AddHandler(handler.MessageHandler)
	runBot(client)
}
```

Error message:
```
  panic: runtime error: invalid memory address or nil pointer dereference
  [signal 0xc0000005 code=0x0 addr=0x18 pc=0x11f8f28]
  
  goroutine 34 [running]:
  github.com/MikeModder/anpan.(*CommandHandler).MessageHandler(0xc00014e2a0, 0xc000126480, 0xc00008e030)
          C:/Users/mrali/go/pkg/mod/github.com/!mike!modder/anpan@v0.0.0-20210425181706-21d818f4ab3f/handler.go:427 +0x4c8
  github.com/bwmarrin/discordgo.messageCreateEventHandler.Handle(0xc000052e60, 0xc000126480, 0x1257260, 0xc00008e030)
          C:/Users/mrali/go/pkg/mod/github.com/bwmarrin/discordgo@v0.23.2/eventhandlers.go:497 +0x52
  created by github.com/bwmarrin/discordgo.(*Session).handle
          C:/Users/mrali/go/pkg/mod/github.com/bwmarrin/discordgo@v0.23.2/event.go:171 +0x118
  exit status 2
```

**Comment by yeah-its-gloria** [comment]
Fixed in commit 841dc5d (Release 8.1.2).
Thanks for the report.

**Comment by haashemi** [comment]
Thanks a lot!
Now I'm using it and it works perfectly.

When I was checking the update, found something that could help to improve the project.

-- it would be awesome to add `SetHelpCommand` and `SetOnErrorFunc` as an argument on `anpan.New`. (force users to use it instead of crashing, it happens if we don't use them)
OR
-- Maybe add some default functions for them (defining some default functions on making handler, the user could update them with using SetX fucntion)
OR
-- Handle them if they were `nil`, just ignoring them with some if/else stuff lol

-- Also, Please write documentation for it. it's so useful, and really needed!

Thanks again, have a nice day! <3
